Documentation/gettingstarted/k8s-install-helm.rst

```diff
@@ -112,15 +112,15 @@ Install Cilium
        .. code-block:: shell-session
 
           AZURE_SUBSCRIPTION_ID=$(az account show --query "id" --output tsv)
-          AZURE_NODE_RESOURCE_GROUP=$(az aks show --resource-group ${RESOURCE_GROUP} --name ${CLUSTER_NAME} --query "nodeResourceGroup" --output tsv)
-          AZURE_SERVICE_PRINCIPAL=$(az ad sp create-for-rbac --scopes /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_NODE_RESOURCE_GROUP} --role Contributor --output json --only-show-errors)
+          AZURE_RESOURCE_GROUP=$(az aks show --resource-group ${RESOURCE_GROUP} --name ${CLUSTER_NAME} --query "nodeResourceGroup" --output tsv)
+          AZURE_SERVICE_PRINCIPAL=$(az ad sp create-for-rbac --scopes /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP} --role Contributor --output json --only-show-errors)
           AZURE_TENANT_ID=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.tenant')
           AZURE_CLIENT_ID=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.appId')
           AZURE_CLIENT_SECRET=$(echo ${AZURE_SERVICE_PRINCIPAL} | jq -r '.password')
 
        .. note::
 
-          The ``AZURE_NODE_RESOURCE_GROUP`` node resource group is *not* the
+          The ``AZURE_RESOURCE_GROUP`` node resource group is *not* the
```

Comment:
The rationale for not using the same terminology is exactly because of what's stated here: the internal node resource group used for Azure IPAM is not the same as the resource group the user provides when creating the cluster.

When I initially made this change, I thought this was a good way to make it clear they are not the same and improve user understanding of what's going on, and still think the same, so I would be inclined not to change it. Happy to hear other opinions on this :)

Reply:
Hi Nicolas and thank you for your attention to my PR.
Although I understand the note on the resource group the final environment variable used in cilium called `AZURE_RESOURCE_GROUP` is filled with Helm value `azure.resourceGroup`
```
$ grep -r -C1 AZURE_RESOURCE_GROUP charts/cilium
charts/cilium/templates/cilium-operator/deployment.yaml-        {{- if .Values.azure.resourceGroup }}
charts/cilium/templates/cilium-operator/deployment.yaml:        - name: AZURE_RESOURCE_GROUP
charts/cilium/templates/cilium-operator/deployment.yaml-          value: {{ .Values.azure.resourceGroup }}
```
If you want to keep emphasization that azure resource group is meant for the Nodes and not the AKS itself i.e. `nodeResourceGroup` field in the json output of `az cli` I suggest to change the underlying deployment env variable as well to keep the names in sync. But that woudln't be just documentation change and may bring more confusion.

I am not fan of Helm so I am trying to patch generated helm using kustomize and that is why I hit this discrepancy.

Reply:
I personally disagree: the Azure IPAM uses `AZURE_RESOURCE_GROUP` because it does not care over which RG it will operate, so the fact the RG passed in to IPAM is an arbitrary top-level RG or the low-level node RG does not matter from its point of view.

On the other hand, when deploying, the administrator should care about which RG they pass to IPAM, because the lower-level RG is more secure (IPAM constrainted to its scope, cannot interfere with APIs outside of the node RG, which might be an issue if the RG containing the AKS cluster also hosts other Azure objects). Yet, this is not a black and white situation because the administrator might consciously decide to use IPAM at an higher level for some reason (other tools interacting with Azure APIs at this level, etc.).

In this sort of situation, I personally tend to err on the side "document properly and leave it to the human to know what they're doing, but still try to make it easy for someone to just copy/paste if they don't care" => hence how the GSG is structured at the moment.

Does this make sense? What is your opinion on this, considering the above?

(Sorry for the delay in answering this.)

Reply:
When I check the source code I find following
https://github.com/cilium/cilium/blob/9a0858e2a70f8103496aa35b6b771ba4c5695128/operator/option/config.go#L193
https://github.com/cilium/cilium/blob/9a0858e2a70f8103496aa35b6b771ba4c5695128/operator/option/config.go#L194

and if I get it correctly, it translates into environment variable `AZURE_RESOURCE_GROUP`, hence it makes sense to use such name even in helm scripts.

Thank you for detailed insight into Azure IPAM.

For me it is confusing the variable used in GSG (getting stared guide?) is not used in the cilium operator itself. However I understand the need to emphasize  that the resource group is different from the one were the **kubernetes service** so I must accept your reasoning. @nbusseneau I let you close the PR.